packages/trader/src/AppV2/Components/TradeParameters/TakeProfit/take-profit.tsx

```diff
@@ -1,26 +1,216 @@
+                            textAlignment='center'
+                            unitLeft={currency}
+                            variant='fill'
+                            value={updated_take_profit_value}
+                        />
+                        {!is_take_profit_enabled && (
+                            <button
+                                className='take-profit__overlay'
+                                onClick={() => onToggleSwitch(true)}
+                                data-testid='dt_take_profit_overlay'
+                            />
+                        )}
+                    </ActionSheet.Content>
+                    <ActionSheet.Footer
+                        alignment='vertical'
+                        primaryAction={{
+                            content: <Localize i18n_default_text='Save' />,
+                            onAction: onSave,
+                        }}
+                        shouldCloseOnPrimaryButtonClick={false}
+                    />
+                </React.Fragment>
+            ),
+        },
+        {
+            id: 2,
+            component: (
+                <ActionSheet.Content className='take-profit__wrapper--definition'>
+                    <Text>
+                        <Localize i18n_default_text='When your profit reaches or exceeds the set amount, your trade will be closed automatically.' />
+                    </Text>
+                    {is_accumulator && (
+                        <SectionMessage
+                            className='take-profit__warning'
+                            message={
+                                <Localize i18n_default_text="Take profit can't be adjusted for ongoing accumulator contracts." />
+                            }
+                            size='sm'
+                            status='info'
+                        />
+                    )}
+                </ActionSheet.Content>
+            ),
+        },
+    ];
+
+    React.useEffect(() => {
+        setIsTakeProfitEnabled(has_take_profit);
+        setUpdatedTakeProfitValue(take_profit);
+
+        return () => clearTimeout(focus_timeout.current);
+    }, [has_take_profit, take_profit]);
+
     return (
-        <TextField
-            variant='fill'
-            readOnly
-            label={<Localize i18n_default_text='Take profit' key={`take-profit${is_minimized ? '-minimized' : ''}`} />}
-            value={take_profit ? `${take_profit} ${getCurrencyDisplayCode(currency)}` : localize('Not set')}
-            className={clsx('trade-params__option', is_minimized && 'trade-params__option--minimized')}
-            disabled={has_open_accu_contract}
-        />
+        <React.Fragment>
+            <TextField
+                variant='fill'
+                readOnly
+                label={
+                    <Localize i18n_default_text='Take profit' key={`take-profit${is_minimized ? '-minimized' : ''}`} />
```

Comment:
Does having the conditional key value here make any difference?

Reply:
There are cases, when minimized and non-minimized trade params (both) exist in DOM, so i don't won't the to have identical key